include/openthread/platform/infra_if.h

```diff
@@ -94,24 +105,22 @@ extern void otPlatInfraIfRecvIcmp6Nd(otInstance *        aInstance,
  * The infra interface driver calls this method to notify OpenThread
  * of the interface state changes.
  *
+ * It is fine for the platform to call to method even when the running state
+ * of the interface isn't changed. In this case, the Routing Manager state is
+ * not affected.
+ *
  * @param[in]  aInstance          The OpenThread instance structure.
  * @param[in]  aInfraIfIndex      The index of the infrastructure interface.
  * @param[in]  aIsRunning         A boolean that indicates whether the infrastructure
  *                                interface is running.
- * @param[in]  aLinkLocalAddress  A pointer to the IPv6 link-local address of the infrastructure
- *                                interface. NULL if the IPv6 link-local address is lost.
  *
  * @retval  OT_ERROR_NONE           Successfully updated the infra interface status.
  * @retval  OT_ERROR_INVALID_STATE  The Routing Manager is not initialized.
  * @retval  OT_ERROR_INVALID_ARGS   The @p aInfraIfIndex doesn't match the infra interface the
- *                                  Routing Manager are initialized with, or the @p aLinkLocalAddress
- *                                  is not a valid IPv6 link-local address.
+ *                                  Routing Manager are initialized with.
  *
  */
-extern otError otPlatInfraIfStateChanged(otInstance *        aInstance,
-                                         uint32_t            aInfraIfIndex,
-                                         bool                aIsRunning,
-                                         const otIp6Address *aLinkLocalAddress);
+extern otError otPlatInfraIfStateChanged(otInstance *aInstance, uint32_t aInfraIfIndex, bool aIsRunning);
```

Comment:
```suggestion
extern otError otPlatInfraIfStateChanged(otInstance *aInstance, uint32_t aInfraIfIndex);
```
Suggest to remove `bool aIsRunning` so that the upper layer has to query whatever information it needs.

It looks strange to me why `aIsRunning` is reported but other information is not.

Reply:
The difference between infra interface status and LLA is that
1. The Routing Manager is notified of the infra interface status passively.
2. The Routing Manager now asks for infra interface LLA initiatively so that we can remove it from this status notification function.